core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -123,6 +126,48 @@
   private SqlFunctions() {
   }
 
+  /** SQL TO_BASE64(string) function. */
+  public static String toBase64(String string) {
+    String base64 = Base64.getEncoder().encodeToString(string.getBytes(UTF_8));
+    StringBuilder str = new StringBuilder(base64.length() + base64.length() / 76);
+    int i;
+    if (base64.length() > 76) {
+      for (i = 0; i + 76 < base64.length(); i += 76) {
+        str.append(base64, i, i + 76);
+        str.append("\n");
+      }
+      str.append(base64, i, base64.length());
+      return str.toString();
+    }
+    return base64;
+  }
+
+  /** SQL TO_BASE64(string) function for binary string. */
+  public static String toBase64(ByteString string) {
+    String base64 = Base64.getEncoder().encodeToString(string.getBytes());
+    StringBuilder str = new StringBuilder(base64.length() + base64.length() / 76);
+    int i;
+    if (base64.length() > 76) {
+      for (i = 0; i + 76 < base64.length(); i += 76) {
```

Comment:
It looks there is a bit of code duplication between `toBase64(String string)` and `toBase64(ByteString)` methods.

Can you use Guava's [Splitter.fixedLength()](https://guava.dev/releases/23.0/api/docs/com/google/common/base/Splitter.html#fixedLength-int-) here ?